### ISSUE TITLE
Owncloud opt-qt5 package is not available in isv:/ownCloud:/desktop/CentOS_6 repository

### ISSUE THREAD

**Issue by er47** [owncloud/client#4400]
Hi,

I try to update owncloud-client.x86_64 0:2.0.2-11.1 to owncloud-client.x86_64 0:2.1.0-7.1 on CentOS6.
I receive following error messages:

Error: Package: owncloud-client-2.1.0-7.1.x86_64 (isv_ownCloud_desktop)
           Requires: opt-qt5-qtbase
Error: Package: owncloud-client-2.1.0-7.1.x86_64 (isv_ownCloud_desktop)
           Requires: opt-qt5-qtwebkit
Error: Package: owncloud-client-2.1.0-7.1.x86_64 (isv_ownCloud_desktop)
           Requires: opt-qt5-qtbase-gui

I checked and there are no opt-qt5-* packages in CentOS 6 repository. Can you please provide the necessary packages. 

Thanks
er

**Comment by UltrasoundSam** [comment]
Hi,

Just like to add, I've got a similar problem for CentOS7, although I'm only missing the qtwebkit package:

Error: Package: owncloud-client-2.1.0-7.1.x86_64 (isv_ownCloud_desktop)
           Requires: opt-qt5-qtwebkit

Thanks!

**Comment by jnweiger** [comment]
**CentOS6**
<pre>
rpm --import http://download.opensuse.org/repositories/isv:/ownCloud:/desktop//CentOS_6/repodata/repomd.xml.key
wget -nv http://download.opensuse.org/repositories/isv:/ownCloud:/desktop/CentOS_6/isv:ownCloud:desktop.repo -O /etc/yum.repos.d/isv:ownCloud:desktop.repo
yum clean all && yum install -y owncloud-client
</pre>

**CentOS7**
<pre>
rpm --import http://download.opensuse.org/repositories/isv:/ownCloud:/desktop//CentOS_7/repodata/repomd.xml.key
wget -nv http://download.opensuse.org/repositories/isv:/ownCloud:/desktop/CentOS_7/isv:ownCloud:desktop.repo -O /etc/yum.repos.d/isv:ownCloud:desktop.repo
yum clean all && yum install -y owncloud-client
</pre>
brings me owncloud-client-2.1.0 including the opt-qt5 dependencies.

**Comment by jnweiger** [comment]
fix deployed, rebuilding 2.1.0 in isv:ownCloud:desktop

**Comment by UltrasoundSam** [comment]
Using the above instructions, all of the dependencies were brought down and I could reinstall ownCloud. However, I've run into another problem. Not sure if I should open another issue, or not, though.

After install, when I try to run ownCloud, it comes up with the following error:

/usr/bin/owncloud: line 2: /opt/qt-5.4/bin/qt*-env.sh: No such file or directory

After checking, there are only two files in  /opt/qt-5.4/bin/ -- owncloud and owncloudcmd. If I comment out the second line in /usr/bin/owncloud, the client starts and everything works fine. I'm just not sure what the second line is doing, and whether it's advisable to just comment it out like that. I tried using a clean install of CentOS7 on a virtual machine, and I have the same problem (and same solution).

**Comment by er47** [comment]
Thank you! Works like a charm. Thanks for the fast fix.

**Comment by fmelaccio** [comment]
Hi,

yesterday I had the same problem on SL6, now installation dependencies are fixed. Thanks for the speed! However the client failed to run the first time with a symbol lookup error in libQtSql.so.4. I noticed that I had this file from the qt4 packages coming with SL6, so I manually installed opt-libqt4-sql from the ownCloud desktop repo and the client ran successfully. Added this for whoever stumbles upon this issue.

**Comment by jnweiger** [comment]
@fmelaccio True. I reproduce the libQtSql.so.4 issue.
Added a dependency for CentOS_6 check out owncloud-client-2.1.0-11.1 or later.

**Comment by ianabc** [comment]
Are the RHEL packages updated at the same time? I'm seeing the problem about missing opt-qt5-qtwebkit on RHEL 7, but if I point at the CentOS 7 repositories I don't get the missing package message.

**Comment by jnweiger** [comment]
Yes, built RHEL_7 and CentOS_7 were built at the same time. But unfortunately the openSUSE unfrastructure only pushed opt-qt5-qtbase and opt-qt5-qttools packages to the mirrors for RHEL_7, not opt-qt5-qtwebkit. I've triggered a rebuild there. HTH.

**Comment by ianabc** [comment]
Yup, that seems to fix it for me on RHEL. Thank you!

**Comment by Gijom** [comment]
Hi, it seems that opt-qt5-qtwebkit is missing (at least this is the error I get) from CenOS_7 clien repo. I used the command mentioned above and still have the same problem.

```
$ yum --disablerepo="*" --enablerepo="isv_ownCloud_desktop" list available                                                                                                  [2994][11:33:01]
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
Available Packages
cmake-debuginfo.x86_64                                                                                         2.8.12.2-2.6                                                                 isv_ownCloud_desktop
libowncloudsync-devel.x86_64                                                                                   2.1.1-2.5                                                                    isv_ownCloud_desktop
libowncloudsync0.x86_64                                                                                        2.1.1-2.5                                                                    isv_ownCloud_desktop
libqtkeychain-debuginfo.x86_64                                                                                 0.5-7.2                                                                      isv_ownCloud_desktop
libqtkeychain-devel.x86_64                                                                                     0.5-7.2                                                                      isv_ownCloud_desktop
opt-libqt5keychain-debuginfo.x86_64                                                                            0.5-9.1                                                                      isv_ownCloud_desktop
opt-libqt5keychain-devel.x86_64                                                                                0.5-9.1                                                                      isv_ownCloud_desktop
opt-libqt5keychain0.x86_64                                                                                     0.5-9.1                                                                      isv_ownCloud_desktop
opt-qt5-qtbase-debuginfo.x86_64                                                                                5.4.2-4.2                                                                    isv_ownCloud_desktop
opt-qt5-qtbase-devel.x86_64                                                                                    5.4.2-4.2                                                                    isv_ownCloud_desktop
opt-qt5-qtbase-examples.x86_64                                                                                 5.4.2-4.2                                                                    isv_ownCloud_desktop
opt-qt5-qtbase-mysql.x86_64                                                                                    5.4.2-4.2                                                                    isv_ownCloud_desktop
opt-qt5-qtbase-odbc.x86_64                                                                                     5.4.2-4.2                                                                    isv_ownCloud_desktop
opt-qt5-qtbase-postgresql.x86_64                                                                               5.4.2-4.2                                                                    isv_ownCloud_desktop
opt-qt5-qtbase-static.x86_64                                                                                   5.4.2-4.2                                                                    isv_ownCloud_desktop
opt-qt5-qttools.x86_64                                                                                         5.4.2-3.1                                                                    isv_ownCloud_desktop
opt-qt5-qttools-common.noarch                                                                                  5.4.2-3.1                                                                    isv_ownCloud_desktop
opt-qt5-qttools-debuginfo.x86_64                                                                               5.4.2-3.1                                                                    isv_ownCloud_desktop
opt-qt5-qttools-devel.x86_64                                                                                   5.4.2-3.1                                                                    isv_ownCloud_desktop
opt-qt5-qttools-examples.x86_64                                                                                5.4.2-3.1                                                                    isv_ownCloud_desktop
opt-qt5-qttools-libs-clucene.x86_64                                                                            5.4.2-3.1                                                                    isv_ownCloud_desktop
opt-qt5-qttools-libs-designer.x86_64                                                                           5.4.2-3.1                                                                    isv_ownCloud_desktop
opt-qt5-qttools-libs-designercomponents.x86_64                                                                 5.4.2-3.1                                                                    isv_ownCloud_desktop
opt-qt5-qttools-libs-help.x86_64                                                                               5.4.2-3.1                                                                    isv_ownCloud_desktop
opt-qt5-qttools-static.x86_64                                                                                  5.4.2-3.1                                                                    isv_ownCloud_desktop
owncloud-client.x86_64                                                                                         2.1.1-2.5                                                                    isv_ownCloud_desktop
owncloud-client-debuginfo.x86_64                                                                               2.1.1-2.5                                                                    isv_ownCloud_desktop
owncloud-client-doc.x86_64                                                                                     2.1.1-2.5                                                                    isv_ownCloud_desktop
owncloud-client-l10n.x86_64                                                                                    2.1.1-2.5                                                                    isv_ownCloud_desktop
owncloud-client-nautilus.x86_64                                                                                2.1.1-2.5                                                                    isv_ownCloud_desktop
owncloud-client-nemo.x86_64
```

**Comment by jnweiger** [comment]
<pre>
rpm --import http://download.opensuse.org/repositories/isv:/ownCloud:/desktop//CentOS_7/repodata/repomd.xml.key
wget -nv http://download.opensuse.org/repositories/isv:/ownCloud:/desktop/CentOS_7/isv:ownCloud:desktop.repo -O /etc/yum.repos.d/isv:ownCloud:desktop.repo

yum --disablerepo="*" --enablerepo="isv_ownCloud_desktop" list available | grep webkit
opt-qt5-qtwebkit.x86_64                        5.4.2-2.3    isv_ownCloud_desktop
opt-qt5-qtwebkit-debuginfo.x86_64              5.4.2-2.3    isv_ownCloud_desktop
opt-qt5-qtwebkit-devel.x86_64                  5.4.2-2.3    isv_ownCloud_desktop
opt-qtwebkit-debuginfo.x86_64                  2.2.3-28.1   isv_ownCloud_desktop
qt5-designer-plugin-webkit.x86_64              5.4.2-3.1    isv_ownCloud_desktop
</pre>

Probably a temporary outage. I can see there was a rebuild finished "build qt5-qtwebkit.spec" at Wed Mar 16 16:13:47 UTC 2016.

**Comment by Gijom** [comment]
Sorry the yum command still doe not list opt-qt5-qtwebkit (even after a `yum clean all`)

Here is my repo file:
[isv_ownCloud_desktop]
name=The ownCloud Desktop Client (CentOS_7)
type=rpm-md
baseurl=http://download.opensuse.org/repositories/isv:/ownCloud:/desktop/CentOS_7/
gpgcheck=1
gpgkey=http://download.opensuse.org/repositories/isv:/ownCloud:/desktop/CentOS_7//repodata/repomd.xml.key
enabled=1

Any idea ?

**Comment by jnweiger** [comment]
Your repo file looks good.

**Comment by apeyser** [comment]
fedora 23:
````
Error: nothing provides opt-qt5-qtwebkit needed by owncloud-client-2.2.4-1.1.x86_64.
nothing provides opt-qt5-qtwebkit needed by owncloud-client-2.2.4-1.1.x86_64.
nothing provides opt-qt5-qtwebkit needed by owncloud-client-2.2.4-1.1.x86_64
````

I was fine until this latest update.

**Comment by ianaharding** [comment]
What the heck?  OwnCloud Client seems to be a bit of a dependency trainwreck.  It installs fine, but is silently missing dependencies which when I try to install them I get stuff like this:

warning: /var/cache/yum/x86_64/7/owncloud64/packages/libqtkeychain0-0.5-7.1.x86_64.rpm: Header V3 DSA/SHA1 Signature, key ID ba684223: NOKEY
Public key for libqtkeychain0-0.5-7.1.x86_64.rpm is not installed
libqtkeychain0-0.5-7.1.x86_64.rpm                                                                                                              |  52 kB  00:00:00     

That key seems not to exists.  Ugh

**Comment by SamuAlfageme** [comment]
Hey @ianaharding! We had some build system turbulence past weeks that might have affected the integrity of the linux packages there. 

We're about to release into the stable channel some new packages that include all the work carried out for https://github.com/owncloud/client/issues/5470, you can help us testing by installing them from the `nightly` repositories: 

https://software.opensuse.org//download.html?project=isv:ownCloud:community:nightly&package=owncloud-client

Try them out and tell us if that worked out for you!
